pandas/core/dtypes/common.py

```diff
@@ -596,12 +596,17 @@ def is_dtype_equal(source, target) -> bool:
     False
     """
     if isinstance(target, str):
-        if not isinstance(source, str):
+        if get_dtype(source) == "string":
```

Comment:
why is this needed?

Reply:
It is directly False, there is no need for further checking for this.

Reply:
the reason i ask is this might be a performance issue here (this code is hit a lot so we care about the code path)

Reply:
I have changed the code to remove this code segment, now there should not be a problem.